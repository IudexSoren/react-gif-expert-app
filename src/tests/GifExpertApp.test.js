import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas de GifExpertApp', () => {

  test('Debe renderizar correctamente el componente', () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar una lista de categorÃ­as', () => {
    const categories = ['One Punch', 'DBZ'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  })


})
