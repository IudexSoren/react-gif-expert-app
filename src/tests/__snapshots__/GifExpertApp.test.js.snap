// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifExpertApp Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="DBZ"
      key="DBZ"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas de GifExpertApp Debe renderizar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
